src/applications/vaos/components/FutureAppointmentsList.jsx

```diff
@@ -26,154 +21,163 @@ import { getVAAppointmentLocationId } from '../services/appointment';
 import ConfirmedAppointmentListItem from './ConfirmedAppointmentListItem';
 import AppointmentRequestListItem from './AppointmentRequestListItem';
 import NoAppointments from './NoAppointments';
+import { resetDataLayer } from '../utils/events';
 
-export class FutureAppointmentsList extends React.Component {
-  componentDidMount() {
-    if (
-      !this.props.showExpressCare &&
-      this.props.futureStatus === FETCH_STATUS.notStarted
-    ) {
-      this.props.fetchFutureAppointments();
-    }
-  }
-
-  render() {
-    const {
-      showPastAppointments,
-      showCancelButton,
-      showScheduleButton,
-      isCernerOnlyPatient,
-      future,
-      futureStatus,
-      facilityData,
-      requestMessages,
-      expressCare,
-    } = this.props;
+function FutureAppointmentsList({
+  showPastAppointments,
+  showCancelButton,
+  showScheduleButton,
+  isCernerOnlyPatient,
+  future,
+  futureStatus,
+  facilityData,
+  requestMessages,
+  expressCare,
+  cancelAppointment,
+  fetchRequestMessages,
+  fetchFutureAppointments,
+  startNewAppointmentFlow,
+}) {
+  useEffect(
+    () => {
+      if (futureStatus === FETCH_STATUS.succeeded) {
+        recordEvent({
+          event: 'nav-tab-click',
+          'tab-text': 'Upcoming',
+          [`${GA_PREFIX}-upcoming-number-of-cards`]: future?.length,
+        });
+        resetDataLayer();
+      }
+    },
+    [futureStatus, future],
+  );
 
-    let content;
+  useEffect(() => {
+    if (!expressCare.enabled && futureStatus === FETCH_STATUS.notStarted) {
```

Comment:
We apparently had a bug where we were double fetching, because we were checking a `showExpressCare` prop that doesn't exist. A bunch of tests broke when I made the error in the other direction, so that's good, I guess.

Reply:
Nice catch